src/pages/ProfilePage.js

```diff
@@ -1,16 +1,128 @@
+
+  const classes = useStyles();
+
+  const handleClickGame = (gameId) => {
+    setRedirect(`/room/${gameId}`);
+  };
+
+  useEffect(() => {
+    function mergeGameData(game, gameData) {
+      const scores = computeState(gameData).scores;
+      const winner = Object.keys(game.users).sort((u1, u2) => {
+        const s1 = scores[u1] || 0;
+        const s2 = scores[u2] || 0;
+        if (s1 !== s2) return s2 - s1;
+        return u1 < u2 ? -1 : 1;
+      })[0];
+      return {
+        ...game,
+        ...gameData,
+        winner: winner,
+        scores: scores,
+      };
+    }
+    async function getGamesData() {
+      const gameReads = [];
+      const gameDataReads = [];
+      for (const gameId of Object.keys(games || {})) {
+        gameReads.push(
+          firebase.database().ref(`games/${gameId}`).once("value")
+        );
+        gameDataReads.push(
+          firebase.database().ref(`gameData/${gameId}`).once("value")
+        );
+      }
+      const gameReadsValues = await Promise.all(gameReads);
+      const gameDataReadsValues = await Promise.all(gameDataReads);
+      if (!loadingGames) {
+        const gamesDataCopy = {};
+        Object.keys(games || {}).forEach((gameId, i) => {
+          const game = gameReadsValues[i].val();
+          const gameData = gameDataReadsValues[i].val();
+          if (game.status === "done") {
+            gamesDataCopy[gameId] = {
+              ...mergeGameData(game, gameData),
+              gameId: gameId,
+            };
+          }
+        });
+        setGamesData(gamesDataCopy);
+      }
+    }
+    getGamesData();
+  }, [games, userId, loadingGames]);
+
+  if (loadingGames) {
+    return <LoadingPage />;
+  }
+  if (redirect) return <Redirect push to={redirect} />;
+
   return (
     <Container>
-      <Paper style={{ padding: 8 }}>
-        <User component={Typography} variant="h4" id={match.params.id} />
+      <Paper style={{ padding: 16 }}>
+        <Grid container className={classes.mainGrid}>
+          <Grid item xs={12} sm={12} md={4} className={classes.usernamePanel}>
+            <ProfileName userId={userId} />
+          </Grid>
+          <Divider
+            orientation="vertical"
+            variant="middle"
+            flexItem
+            className={classes.divider}
+          />
+          <Grid item xs={12} sm={12} style={{ flex: 1 }} p={1}>
+            <div style={{ display: "flex" }}>
+              <Typography variant="overline"> Statistics</Typography>
+              <EqualizerIcon />
+            </div>
+            <UserStatistics
+              userId={userId}
+              gamesData={gamesData}
+              loadingGames={loadingGames}
+            />
+          </Grid>
+          <Grid item xs={12} sm={12} md={12}>
+            <Typography variant="overline"> Finished Games</Typography>
+            <ProfileGamesTable
+              userId={userId}
+              handleClickGame={handleClickGame}
```

Comment:
Rather than redirects and `handleClickGame` here, it would probably just be simpler to remove all the redirect code and just put anchor tags/link components in `ProfileGamesTable`. See my comments there for more details.